openstreetmap-carto.lua

```diff
@@ -378,46 +369,24 @@ function filter_tags_relation_member (keyvalues, keyvaluemembers, roles, memberc
    return filter, keyvalues, membersuperseded, linestring, polygon, roads
 end
 
--- Check if an object with given tags should be treated as polygon
-function haspolygontags (tags)
-   -- Treat objects tagged as area=yes, area=1, or area=true as polygon,
-   -- and treat objects tagged as area=no, area=0, or area=false not as polygon
-   if ((tags["area"] == "yes") or (tags["area"] == "1") or (tags["area"] == "true")) then
-      return 1
-   elseif ((tags["area"] == "no") or (tags["area"] == "0") or (tags["area"] == "false")) then
-      return 0
-   end
-
-   -- Treat objects with a key in polygon_keys as polygon
-   local haspolytags = 0
-   for i,k in ipairs(polygon_keys) do
-      if tags[k] then
-         local polygontag = 1
-         -- However, if the key/value combination occurs in linestring_values, do not treat the object as polygon
-         for index,tag in pairs(linestring_values) do
-            if k == tag[1] and tags[k] == tag[2] then
-               polygontag = 0
-               break
-            end
-         end
-         if polygontag == 1 then
-            haspolytags = 1
-            break
-         end
-      end
+--- Check if an object with given tags should be treated as polygon
+-- @param tags OSM tags
+-- @return 1 if area, 0 if linear
+function isarea (tags)
+   -- Treat objects tagged as area=yes polygon, other area as no
+   if tags["area"] then
+      return tags["area"] == "yes" and 1 or 0
    end
 
-   -- Treat objects with a key/value combination in polygon_values as polygon
-   if haspolytags == 0 then
-      for index,tag in pairs(polygon_values) do
-         if tags[tag[1]] == tag[2] then
-            haspolytags = 1
-            break
-         end
+   -- Search through object's tags
+   for k, v in pairs(tags) do
+      -- Check if it has a polygon key and not a linestring override, or a polygon k=v
+      if (polygon_keys[k] and not (linestring_values[k] and linestring_values[k][v])) or
```

Comment:
Hi @pnorman, do you agree this needs changing? Or am I overlooking something and is it correct as is?